boltdb/adaptor_test.go

```diff
@@ -0,0 +1,221 @@
+package boltdb_test
+
+import (
+	"context"
+	"errors"
+
+	"github.com/dogmatiq/dogma"
+	"github.com/dogmatiq/enginekit/fixtures"
+	. "github.com/dogmatiq/projectionkit/boltdb"
+	"github.com/dogmatiq/projectionkit/boltdb/internal"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"go.etcd.io/bbolt"
+)
+
+var _ = Describe("MessageHandler Adaptor", func() {
+	var (
+		mh *internal.MessageHandlerMock
+		db *internal.TempDB
+	)
+
+	BeforeEach(func() {
+		mh = &internal.MessageHandlerMock{}
+		mh.ConfigureCall = func(c dogma.ProjectionConfigurer) {
+			c.Identity("<projection>", "<key>")
+		}
+		db = internal.NewTempDB()
+	})
+
+	AfterEach(func() {
+		db.Close()
+	})
+
+	Context("HandleEvent method", func() {
+
+		It("executes as expected", func() {
+			ctx := context.TODO()
+			a := New(db.DB, mh)
+			r := []byte("<resource>")
+			c := []byte{}
+			n := []byte("<version 01>")
+
+			By("peristing the correct initial version")
+
+			ok, err := a.HandleEvent(
+				ctx,
+				r,
+				c,
+				n,
+				nil,
+				fixtures.MessageA1,
+			)
+
+			Expect(ok).Should(BeTrue())
+			Expect(err).ShouldNot(HaveOccurred())
+
+			v, err := a.ResourceVersion(ctx, r)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(v).To(Equal(n))
+
+			By("peristing the next correct version")
+
+			c = n
+			n = []byte("<version 02>")
+
+			ok, err = a.HandleEvent(
+				ctx,
+				r,
+				c,
+				n,
+				nil,
+				fixtures.MessageA2,
+			)
+
+			Expect(ok).Should(BeTrue())
+			Expect(err).ShouldNot(HaveOccurred())
+
+			v, err = a.ResourceVersion(ctx, r)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(v).To(Equal(n))
+
+			By("deleting resource if the next version is empty")
+
+			c = n
+			n = []byte{}
+
+			ok, err = a.HandleEvent(
+				ctx,
+				r,
+				c,
+				nil,
+				nil,
+				fixtures.MessageA3,
+			)
+			Expect(ok).Should(BeTrue())
+			Expect(err).ShouldNot(HaveOccurred())
+
+			v, err = a.ResourceVersion(ctx, r)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(v).To(BeNil())
+		})
+
+		It("returns an error if MessageHandler's HandleEvent method fails", func() {
+			ctx := context.TODO()
+			a := New(db.DB, mh)
+			r := []byte("<resource>")
+			c := []byte{}
+			n := []byte("<version 01>")
```

Comment:
I think it's clearer to pass these `[]byte{...}` literals directly to the function, rather than assigning them to variables first, especially considering they are not relevant to this exact test case.

Reply:
Changed in the latest commits to using literals as arguments to the functions.